File: factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/commands/Runner.java
package org.factcast.schema.registry.cli.commands;

import org.factcast.schema.registry.cli.Application;
import org.springframework.boot.*;
import org.springframework.stereotype.Component;
import picocli.CommandLine;

@Component
public class Runner implements CommandLineRunner, ExitCodeGenerator {

  private Application command;
  private CommandLine.IFactory factory; // auto-configured to inject PicocliSpringFactory

  private int exitCode;

  public Runner(Application command, CommandLine.IFactory factory) {
    this.command = command;
    this.factory = factory;
    System.out.println("oink runner");
  }

  @Override
  public void run(String... args) throws Exception {
    System.out.println("running " + args);
    exitCode = new CommandLine(command, factory).execute(args);
  }

  @Override
  public int getExitCode() {
    return exitCode;
  }
}

File: factcast-schema-registry-cli/src/main/kotlin/org/factcast/schema/registry/cli/commands/Runner.kt
package org.factcast.schema.registry.cli.commands

import org.factcast.schema.registry.cli.Application
import org.springframework.boot.CommandLineRunner
import org.springframework.boot.ExitCodeGenerator
import org.springframework.stereotype.Component
import picocli.CommandLine
import picocli.CommandLine.IFactory

@Component
class Runner(
    private val command: Application, // auto-configured to inject PicocliSpringFactory
    private val factory: IFactory
) : CommandLineRunner,
    ExitCodeGenerator {
    private var exitCode: Int = 0

    @Throws(Exception::class)
    override fun run(vararg args: String) {
        this.exitCode = CommandLine(command, factory).execute(*args)
    }

    override fun getExitCode(): Int = this.exitCode

}
